src/core/document.js

```diff
@@ -935,6 +938,65 @@ class PDFDocument {
       ? this.catalog.cleanup(manuallyTriggered)
       : clearPrimitiveCaches();
   }
+
+  _collectFieldObjects(name, fieldRef, promises) {
+    const field = this.xref.fetchIfRef(fieldRef);
+    if (field.has("T")) {
+      const partName = stringToPDFString(field.get("T"));
+      if (name === "") {
+        name = partName;
+      } else {
+        name = `${name}.${partName}`;
+      }
+    }
+
+    if (!(name in promises)) {
+      promises[name] = [];
+    }
+    promises[name].push(
+      AnnotationFactory.create(
+        this.xref,
+        fieldRef,
+        this.pdfManager,
+        this._localIdFactory
+      )
+        .then(annotation => annotation.getFieldObject())
+        .catch(function (reason) {
+          warn(`_collectFieldObjects: "${reason}".`);
+          return null;
+        })
+    );
+
+    if (field.has("Kids")) {
+      const kids = field.get("Kids");
+      for (const kid of kids) {
+        this._collectFieldObjects(name, kid, promises);
+      }
+    }
+  }
+
+  async getFieldObjects() {
+    const formInfo = this.formInfo;
+    const allFields = Object.create(null);
+    if (!formInfo.hasAcroForm) {
+      return allFields;
```

Comment:
Would it not make sense to return `null` instead in this case, since that's consistent with most existing API-methods!?